### PR TITLE
Deployment improvements

### PR DESCRIPTION
**Pull request by marksteve**
- Added step to prune unused docker images/containers
- Added step to notify deployment success
- Use build id as tag to ensure proper images are pulled

### CODE REVIEW

File: deploy/cloudbuild.yml [hunk 1]
```diff
@@ -128,6 +146,7 @@ steps:
   - 'DOCKER_HOST=$_HOST:2376'
   - DOCKER_TLS_VERIFY=1
   - DOCKER_CERT_PATH=.
+  - 'BUILD_ID=$BUILD_ID'
```
Comment (syk0saje) [inline deploy/cloudbuild.yml]: Is it just the ones with `$` references that we single-quote?

File: deploy/docker-compose.yml [hunk 2]
```diff
@@ -11,14 +11,14 @@ services:
     restart: unless-stopped
 
   python:
-    image: gcr.io/linksight-208514/python:latest
+    image: gcr.io/linksight-208514/python:${BUILD_ID}
```
Comment (syk0saje) [inline deploy/docker-compose.yml]: Is there a reason some of these are braced and some are not? Would be nice if we had more consistency.

Comment (marksteve) [reply deploy/docker-compose.yml]: https://docs.docker.com/compose/environment-variables/#configure-compose-using-environment-variables

File: deploy/cloudbuild.yml (hunk 1 above)
Comment (marksteve) [reply deploy/cloudbuild.yml]: Yep

**Pull request merged**